pyvista/plotting/lights.py

```diff
@@ -0,0 +1,465 @@
+        on the type of the light.
+
+    color : list or tuple, optional
+        The color of the light. The ambient, diffuse and specular colors will
+        all be set to this color in creation.
+
+    light_type : string or int, optional
+        The type of the light. If a string, one of ``'headlight'``,
+        ``'camera light'`` or ``'scene light'``. If an int, one of 1, 2 or 3,
+        respectively. The class constants ``Light.HEADLIGHT``, ``Light.CAMERA_LIGHT``
+        and ``Light.SCENE_LIGHT`` are also available, respectively.
+
+        A headlight is attached to the camera, looking at its focal point along
+        the axis of the camera.
+        A camera light also moves with the camera, but it can occupy a general
+        position with respect to it.
+        A scene light is stationary with respect to the scene, as it does not
+        follow the camera. This is the default.
+    """
+
+    # TODO: better/more explanation for ``position``?
+
+    # pull in light type enum values as class constants
+    HEADLIGHT = LightType.HEADLIGHT
+    CAMERA_LIGHT = LightType.CAMERA_LIGHT
+    SCENE_LIGHT = LightType.SCENE_LIGHT
+    # TODO: is this design OK?
+
+    def __init__(self, position=None, color=None, light_type='scene light'):
+        """Initialize the light."""
+        super().__init__()
+
+        if position:
+            self.position = position
+
+        if color:
+            self.ambient_color = color
+            self.diffuse_color = color
+            self.specular_color = color
+            # TODO: make this a single call if that's implemented
+
+        if isinstance(light_type, str):
+            # be forgiving: ignore spaces and case
+            light_type_orig = light_type
+            type_normalized = light_type.replace(' ', '').lower()
+            mapping = {'headlight': LightType.HEADLIGHT,
+                       'cameralight': LightType.CAMERA_LIGHT,
+                       'scenelight': LightType.SCENE_LIGHT,
+                      }
+            light_type = mapping.get(type_normalized)
+            if light_type is None:
+                raise ValueError(f'Invalid ``light_type`` "{light_type_orig}"')
+        elif not isinstance(light_type, int):
+            raise TypeError('Parameter ``light_type`` must be int or str,'
+                            f' not {type(light_type)}.')
+        # LightType is an int subclass
+
+        self.light_type = light_type
+
+        # TODO: default Positional state?
+        # TODO: should setting attenuation and cone angle automatically switch to positional?
+        # TODO: should color and point and direction_angle have more flexible signatures? (only for non-properties)
+        # TODO: ndarray type and shape and size checking for color and point
+
+    def __repr__(self):
+        """Print a repr specifying the id of the light and the vtkLight it wraps."""
+        return (f'<pyvista.Light ({self.light_type}) at {hex(id(self))}>')
```

Comment:
```suggestion
        return (f'<{self.__class__.__name__} ({self.light_type}) at {hex(id(self))}>')
```

Reply:
Good catch, thanks! I originally implemented the `repr` for a composition-based approach. In this case I would have stressed that this is a `pyvista.Light` which wraps a `vtkLight`. This is no longer needed because we aren't wrapping anything.

Also, you can see the obsolete docstring for the `__repr__` which mentions a `vtkLight` being wrapped. Obviously this is nonsense now. Thanks.